modules/core/src/main/scala/iota/internal/Recursion.scala

```diff
@@ -1,85 +0,0 @@
-/*
- * Copyright 2016-2017 47 Degrees, LLC. <http://www.47deg.com>
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package iota
-package internal
-
-import cats.Applicative
-import cats.Functor
-import cats.Monad
-import cats.Traverse
-
-private[internal] object Recursion {
-
-  // The following code is based on Matryoshka
```

Comment:
Do they have a copying or proper way to credit them in their project that we should follow beyond a simple link?